### PR TITLE
Integrate App Check with Firebase umbrella pod 

### PR DESCRIPTION
**Pull request by maksymmalyhin**
- App Check added to Firebase.h
- App Check added Firebase pod
- App Check added to FirebaseManifest.swift
- App Check adoption of Firebase version
- App Check added to Dangerfile 
- disable tests for `pod lib lint` with `--use-libraries` and `--use-modular-headers` options

### CODE REVIEW

**Review by morganchen12 — APPROVED** [review]
LGTM, but please wait for other approvers before merging.

**Review by paulb777 — APPROVED** [review]
LGTM!

Can be done separately - appcheck should also be added to the GHA dashboard in https://github.com/firebase/firebase-ios-sdk/blob/master/README.md

**Comment by maksymmalyhin** [comment]
> LGTM!
> 
> Can be done separately - appcheck should also be added to the GHA dashboard in https://github.com/firebase/firebase-ios-sdk/blob/master/README.md

Will do in a separate PR, thanks!

**Pull request merged**
